### DISCUSSION TITLE
Create “image” for ImageDataset

### DISCUSSION THREAD

**Discussion by yarkinslav** [Project-MONAI/MONAI#4096] [Q&A] [answered]
I am trying to upload “image” with 8 channels for each subject (8,294,300,270) (1 is for T1w and 7 are for extracted Feature maps), to do this I just stack volumes to nn.Stack. I got following question:
- What should be the values of key “image”? Referencing doc, it should be paths to nifti images, but I got tensors (3,294,300,270). How it could be uploaded to ImageDataset?

**Comment by Nic-Ma** [comment]
Hi @relyativist ,

Thanks for your interest here.
I would suggest you to use the `dict` based transforms and dataset:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/dataset.py#L49
So you can put any data in the dict with "image" key.

Thanks.

**Reply by Nic-Ma** [reply]
You can refer to this example: https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/torch/unet_training_dict.py#L86.

Thanks.

**Comment by yarkinslav** [comment, marked as answer]
Hey @Nic-Ma, thanks for quick response. I am being unclear about details. What I was asking is how data with dict can be uploaded to Dataset?
I have input list of key, "image" and "seg" pair. With multiple "image" channels:

```
[{'image': {'t1': '8.nii.gz',
   'curv': 'curv.nii.gz',
   'sulc': 'sulc.nii.gz',
   'contrast1: 'contrast1.nii.gz',
   'contrast2': 'contrast2.nii.gz',
   'glcm': 'glcm.nii.gz',
   'contrast3': 'contrast3.nii.gz',
   'contrast4': 'contrast4.nii.gz'},
  'label': 'label_8.nii.gz'},
  ...
  {}]
```
I tried to follow https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/torch/unet_training_dict.py#L86, but stuck on  `check_ds = monai.data.Dataset(data=train_files, transform=train_transforms)`, as i get multiple maps for 'image'.

**Reply by Nic-Ma** [reply]
Hi @relyativist ,

I see, you put images as `dict in dict` so `Dataset` can't support it so far. Please try with `list in dict` instead, something like:
```
[{'image': ['8.nii.gz', 'curv.nii.gz', 'sulc.nii.gz', 'contrast1.nii.gz', 'contrast2.nii.gz', 'glcm.nii.gz', 'contrast3.nii.gz', 'contrast4.nii.gz'],
  'label': 'label_8.nii.gz'},
  ...
  {}]
```

Thanks.

**Reply by yarkinslav** [reply]
@Nic-Ma Thanks, That worked!

**Reply by yarkinslav** [reply]
@Nic-Ma Little detail I missed, if I need to make Transform on particular channel, e.g. 'curv.nii.gz', 'sulc.nii.gz',, for example:
```
im_trans = Compose(
    [
        ScaleIntensityd(keys=['curv.nii.gz', 'sulc.nii.gz'], minv=0.0, maxv=1.0, channel_wise=True),
        LoadImaged(keys=["image", "label"]),
        Resized(keys=['curv.nii.gz', 'sulc.nii.gz'], spatial_size=(241)),
    ]
)
```
It's seems is will throw an error, because there is no such keys in this dict, as they are values of key 'image'. Is it possible to apply transform on particular channels?

**Reply by Nic-Ma** [reply]
Hi @relyativist ,

If you resize images only on particular channels, you can't stack them together as they have different spatial shape anymore.
If you don't want to stack images, you can save every channel with a key in the dict directly, then apply `resize` transform on expected keys, like:
```
[
  {
   't1': '8.nii.gz',
   'curv': 'curv.nii.gz',
   'sulc': 'sulc.nii.gz',
   'contrast1: 'contrast1.nii.gz',
   'contrast2': 'contrast2.nii.gz',
   'glcm': 'glcm.nii.gz',
   'contrast3': 'contrast3.nii.gz',
   'contrast4': 'contrast4.nii.gz',
   'label': 'label_8.nii.gz'
  },
  ...
  {}
]
```

Thanks.

**Reply by yarkinslav** [reply]
@Nic-Ma Thanks

> If you resize images only on particular channels, you can't stack them together as they have different spatial shape anymore. If you don't want to stack images, you can save every channel with a key in the dict directly, then apply `resize` transform on expected keys, like:

Is it possible to stack each key to create dataset, after transform separate keys?

**Reply by Nic-Ma** [reply]
Yes, if the transformed images have same spatial shape, you can concat them with `ConcatItemsd` transform:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/utility/dictionary.py#L951

Thanks.
